src/common/calendar/TimePickerCalendar.js

```diff
@@ -186,8 +190,19 @@ class TimePickerCalendar extends Component {
         );
       }
 
-      selectable = calendarUtils.getMinPeriodTimeRange(resource, selected.start, selected.end);
       // Make sure selected time will always bigger than min period
+      selectable = calendarUtils.getMinPeriodTimeRange(resource, selected.start, selected.end);
+
+      // Move the selection to nearest available slot
+      if (!this.isSelectionValid(selectable)) {
+        const nearestSlot = this.getNearestValidSlot(selectable, resource);
+        selectable = nearestSlot[0];
+        const nearestSlotIsValid = nearestSlot[1];
+        nearestSlotIsValid && createNotification(
```

Comment:
Hmm, why not `if (nearestSlotIsValid) ...` ?

Reply:
Wanted to save some lines with short circuit which would be same as `if (nearestSlotIsValid)...` .

Reply:
I agree :) 
Changed to `if (...)`